manifests/security/configure_sshd.pp

```diff
@@ -3,7 +3,7 @@
 # @param port            The port to configure sshd to listen on. If undef, we leave it to sshd to decide.
 class sunet::security::configure_sshd (
   Boolean $configure_sftp = true,
-  Optional[Integer] $port = hiera('sunet_ssh_daemon_port', undef),
+  Optional[Integer] $port = lookup('sunet_ssh_daemon_port', Optional[integer], undef,  undef),
```

Comment:
Is this a valid syntax?

Reply:
Yes. As far as I can tell, and as my testing can conclude.

https://www.puppet.com/docs/puppet/5.5/hiera_automatic.html#puppet_lookup

 lookup( <NAME>, [<VALUE TYPE>], [<MERGE BEHAVIOR>], [<DEFAULT VALUE>] ) 
<VALUE TYPE> (data Type) - A data type that must match the retrieved value; if not, the lookup (and catalog compilation) will fail. Defaults to Data which accepts any normal value.